src/0x.ts

```diff
@@ -254,4 +255,45 @@ export class ZeroEx {
         }
         return ecSignature;
     }
+    /**
+     * Returns the ethereum addresses of all available exchange contracts
+     * on the network that the provided web3 instance is connected to
+     * @return  The ethereum addresses of all available exchange contracts.
+     */
+    public async getAvailableExchangeContractAddressesAsync(): Promise<string[]> {
+        const networkId = await this._web3Wrapper.getNetworkIdIfExistsAsync();
+        if (_.isUndefined(networkId)) {
+            return [];
+        } else {
+            const exchangeArtifacts = _.values(ExchangeArtifactsByName);
+            const networkSpecificExchangeArtifacts = _.compact(_.map(
+                exchangeArtifacts, exchangeArtifact => exchangeArtifact.networks[networkId]));
+            const exchangeAddresses = _.map(
+                networkSpecificExchangeArtifacts,
+                networkSpecificExchangeArtifact => networkSpecificExchangeArtifact.address,
+            );
+            return exchangeAddresses;
+        }
+    }
+    /**
+     * Returns the ethereum addresses of all available exchange contracts
+     * on the network that the provided web3 instance is connected to
+     * that are currently authorized on the Proxy contract
+     * @return  The ethereum addresses of all available and authorized exchange contract.
```

Comment:
Returns the ethereum addresses of all available exchange contracts
on the network that the supplied web3 provider is connected to
that are currently authorized by the Proxy contract